### PR TITLE
Fixes #16537 - Tests to ensure play_roles work

### PR DESCRIPTION
**Pull request by dLobatog**
Includes:
  - Fixes for rubocop errors
  - Add foreman_ansible_core tests
  - Fixes and tests for proxy selector
  - Controller tests

### CODE REVIEW

File: lib/foreman_ansible_core/playbook_runner.rb [hunk 1]
```diff
@@ -72,19 +71,27 @@ def ensure_directory(path)
...
         @working_dir = Dir.mktmpdir
         @tmp_working_dir = true
       end
     end
+
+    def initialize_ansible_dir(ansible_dir)
+      if File.exist?(ansible_dir)
+        @ansible_dir = ansible_dir
+      else
+        raise "Ansible dir #{@ansible_dir} does not exist"
```
Comment (iNecas) [inline lib/foreman_ansible_core/playbook_runner.rb]: `s/@ansible_dir/ansible_dir`

File: test/support/foreman_tasks/task.rb [hunk 2]
```diff
@@ -0,0 +1,47 @@
+module Support
+  module ForemanTasks
+    module Task
+      def stub_tasks!
```
Comment (iNecas) [inline test/support/foreman_tasks/task.rb]: Seems like a copy-paste from https://github.com/Katello/katello/blob/master/test/support/foreman_tasks/task.rb, probably right time to extract this to foreman-tasks, to be reusable across projects.

**Comment by iNecas** [comment]
You mentioned "Fixes and tests for proxy selector", what was broken there?

File: test/support/foreman_tasks/task.rb (hunk 2 above)
Comment (dLobatog) [inline test/support/foreman_tasks/task.rb]: Indeed

**Comment by dLobatog** [comment]
@iNecas Nothing, I think I wrote the comment when https://github.com/theforeman/foreman/pull/3824 was still a problem. The only functional change of this PR is adding 'error' messages to the controllers. I'm still not sure if we should cast `:not_defined` to `:not_available`.. it seems rather annoying if someone has a proxy in Organization A with Ansible, and users in Organization B (which doesn't have a proxy nor want it) are forced to install one now

**Comment by iNecas** [comment]
ACK

**Comment by dLobatog** [comment]
Last commit before releasing this. Thanks @iNecas !

**Pull request merged**